rust/arrow/src/datatypes.rs

```diff
@@ -1142,6 +1142,44 @@ impl DataType {
                 | Float64
         )
     }
+
+    /// Compares this data type with another data type only based on the data type
+    /// including nested data types, but not based on other values.
+    pub fn cmp_type(&self, other: &Self) -> bool {
```

Comment:
maybe `equals` instead of compare?

Reply:
True, `cmp()` usually returns `Ordering`.

Reply:
renamed it